src/api/code-challenge/content-types/code-challenge/lifecycles.js

```diff
@@ -24,46 +28,171 @@ const getInternalLabel = async (event) => {
   return `${lesson.name} -- name`;
 };
 
-// runMetaTestsNeed more specific name later, maybe use "validateInternalTests"
-async function runMetaTests(eventTests, eventMetaTests) {
+const iterateMetaTests = async (eventTests, eventMetaTests, challengeLabel) => {
   const [internalTests, metaTests] = await Promise.all([
     getInternalTests(eventTests),
     getMetaTests(eventMetaTests),
   ]);
+  console.log("<metaTests.length>", metaTests.length);
+  console.log("<metaTests>", metaTests);
+  console.log("<code-challenge>", challengeLabel);
+  console.log("<internalTests>: ", internalTests);
+  for (let i = 0; i < metaTests.length; i++) {
+    runInternalTests(
+      internalTests,
+      metaTests[i].caseCode,
+      challengeLabel,
+      metaTests[i].label,
+      metaTests[i].id,
+      metaTests[i].passes
+    );
+  }
+};
+
+const executeTests = async (
+  metaCaseCode,
+  internalTests,
+  metaLabel,
+  challengeLabel,
+  metaTestId,
+  evalResultShouldBe
+) => {
+  return pMap(
+    internalTests?.filter(removeEmpty) || [],
+    async (internalTest) => {
+      const internalTestLabel = internalTest.label;
+      const internalTestCode = internalTest.internalTestCode;
+      const internalTestId = internalTest.id;
+      // console.log("<internalTest pMap> ", internalTest);
+
+      const newTest = {
+        metaLabel,
+        description: null,
+        userPassed: null,
+        resultType: null,
+        evalResult: null,
+        evalResultShouldBe,
+        internalTestLabel,
+        challengeLabel,
+        metaTestId,
+        internalTestId,
+        metaCaseCode,
+        internalTestCode,
+      };
+      // console.log("newTest name: ", newTest);
+
+      const { userPassed, error, description, resultType, evalResult } =
+        await runTestEvaluator({
+          metaCaseCode,
+          internalTestCode,
+          metaLabel,
+          challengeLabel,
+          metaTestId,
+          internalTest,
+          evalResultShouldBe,
+        });
+      newTest.description = description;
+      newTest.resultType = resultType;
+      newTest.evalResult = evalResult;
+      newTest.userPassed = userPassed;
```

Comment:
There is a more elegant way to do this. Something like:

```
const testEvaluatorResults =
        await runTestEvaluator({
          metaCaseCode,
          internalTestCode,
          metaLabel,
          challengeLabel,
          metaTestId,
          internalTest,
          evalResultShouldBe,
        });
newTest = {...newTest, ...pick(testEvaluatorResults, ['description', 'resultType', 'evalResult', 'userPassed'])}
```
The pick function can be imported from lodash

Reply:
Similarly you can include the below error conditional with something like
```
{...newTest, ...pick(testEvaluatorResults, ['description', 'resultType', 'evalResult', 'userPassed']), ...(error ? {error: {message, stack}} : {})}
```
And then, to make things even better you can get rid of the initial definition of `newTest`  and just wrap everything into one return statement